sdk/ai/azure-ai-inference/azure/ai/inference/prompts/_utils.py

```diff
@@ -5,12 +5,18 @@
 # mypy: disable-error-code="import-untyped,return-value"
 # pylint: disable=line-too-long,R,wrong-import-order,global-variable-not-assigned)
 import json
+import os
 import re
 import sys
-import yaml
-from typing import Any, Dict, Union
+from typing import Any, Dict
 from pathlib import Path
 
+try:
+    import yaml  # type: ignore
+except ImportError as exc:
+    raise ImportError("Please install pyyaml to use this function. Run `pip install pyyaml`.") from exc
+
```

Comment:
Shouldn't this be moved to inside the method `def parse(contents):` where the yaml package is used? Otherwise, doesn't this code always get executed? I'm not sure when code at the top of this file gets executed.

Reply:
This line of code will be executed when people tries to use any thing from sub-package prompts. But yes, it's good to move it to somewhere easier to control, let me do it